twofactor_privacyidea/js/settings-admin.js

```diff
@@ -61,7 +61,7 @@ $(document).ready(function () {
 
     /* Activate privacyIDEA */
     getValue("piactive", function(piactive) {
-        $("#piSettings #piactive").prop('checked', piactive !== "0");
+        $("#piSettings #piactive").prop('checked', piactive === "1");
```

Comment:
This will deactivate old owncloud app configuration, because the do not have the ``piactive``.
But they are "active" anyway!
This is not backward compatible.

Reply:
Hmm, I'm still not sure about this. I think this change doesn't affect the functionality of the app at all: In fact, in [version 2.3](https://github.com/NetKnights-GmbH/privacyidea-owncloud-app/blob/v2.3/twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php#L325) of the app, the provider actually checks for ``=== "1"``, which doesn't seem backwards-compatible.
I'll re-check this, maybe we can talk about this next week or so :)

Reply:
Ah ok, it only checks the checkbox in the settings according to the database content, right?
So this will only take effect, if the admin visits the config page...

Reply:
Exactly. And even if the admin visits the config page, this shouldn't change anything.

I'd really like to check this does not break backwards-compatibility. But it turns out to be a bit complicated, as 2.2 does not support ownCloud 10, but 2.3 does not support 9.1 (I think?)

Reply:
So we should check for backward compat to 2.3.
...and we add a RELESE.NOTE

Reply:
So, #16 is about a discrepancy of admin panel and provider in app version 2.3: Right after installation, the admin panel makes the impression that 2FA is enabled, but the provider doesn't actually enforce it. Let's have a look at the relevant lines **in app version 2.3**:
* in the admin panel: https://github.com/NetKnights-GmbH/privacyidea-owncloud-app/blob/28a61d474cf44de5b85cff16e0a7d65f9687f725/twofactor_privacyidea/js/settings-admin.js#L61
* in the provider: https://github.com/NetKnights-GmbH/privacyidea-owncloud-app/blob/28a61d474cf44de5b85cff16e0a7d65f9687f725/twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php#L325

So right after installation, the ``piactive`` setting does not exist, i.e. it is treated as being empty. The provider checks for ``piactive == "1"``, which is correct. The admin panel, however, checks for ``piactive != "0"``, which is true because ``piactive`` is empty! This is because the checkbox is displayed as checked, even though 2FA is actually not enabled.

So I think this PR just makes the provider and the admin panel consistent, i.e. **with this PR** we have
* in the admin panel: https://github.com/NetKnights-GmbH/privacyidea-owncloud-app/blob/cffff3425627eff912f2342c45910334fd939f22/twofactor_privacyidea/js/settings-admin.js#L64
* and in the provider: https://github.com/NetKnights-GmbH/privacyidea-owncloud-app/blob/cffff3425627eff912f2342c45910334fd939f22/twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php#L327

Thus, I think this PR is fully backwards-compatible: If we assume that someone has enabled 2FA in version 2.3, the ``piactive`` setting is ``"1"``. If the user now updates to 2.4 with this PR, this won't change this setting at all.

@cornelinux What do you think?

Reply:
I think in very early versions there was no setting ``piactive``. It simply did not exist!
2FA was active, as soon as it was installed.

This is why we would probably have to check ``piactive !== "0"`` to verify if 2FA is activated.
It would be active if piactive was "1" or if piactive would not exist at all.

So should be change both lines to check for

    piactive !== "0"

?

Reply:
Apparently, ``piactive`` was introduced in 2.3: https://github.com/NetKnights-GmbH/privacyidea-owncloud-app/commit/864afd60d8cafb5c6bbcead6e44ef8e3443af426. So the upgrade from 2.2 to 2.3 is actually backwards incompatible, as 2FA would be inactive after the upgrade!

I think using ``piactive !== "0"`` in provider and admin panel would be problematic because this would mean that 2FA is active immediately after installing the app.

So, I think using ``piactive == "1"`` makes most sense if we consider fresh installations. As for updates:
* **2.3 -> 2.4**: anyone who has 2FA enabled has ``piactive ==="1"``, so 2FA is still enabled after the update.
* **2.2 -> 2.4**: this is indeed a problematic update path, as 2FA would be disabled after the update. Maybe we should add a special case to handle this update path?

Reply:
> So, I think using piactive == "1" makes most sense if we consider fresh installations.
Totally true.